lib/core/src/display/shape/glyph/system.rs

```diff
@@ -0,0 +1,231 @@
+}
+
+impl Glyph {
+    /// Glyph color attribute accessor.
+    pub fn color(&self) -> Attribute<Vector4<f32>> {
+        self.color_attr.clone_ref()
+    }
+
+    /// Change the displayed character.
+    pub fn set_glyph(&mut self, ch:char, fonts:&mut FontRegistry) {
+        let font       = fonts.get_render_info(self.font_id);
+        let glyph_info = font.get_glyph_info(ch);
+        self.msdf_index_attr.set(glyph_info.msdf_texture_glyph_id as f32);
+        self.update_msdf_texture(fonts);
+    }
+
+    fn update_msdf_texture(&mut self, fonts:&mut FontRegistry) {
+        let font = fonts.get_render_info(self.font_id);
+        let texture_changed = self.msdf_uniform.with_content(|texture| {
+            texture.storage().height != font.msdf_texture().rows() as i32
+        });
+        if texture_changed {
+            let msdf_texture = font.msdf_texture();
+            let data         = msdf_texture.data.as_slice();
+            let width        = MsdfTexture::WIDTH         as i32;
+            let height       = msdf_texture.rows() as i32;
+            let texture = Texture::<GpuOnly,Rgb,u8>::new(&self.context,(width,height));
+            texture.reload_with_content(data);
+            self.msdf_uniform.set(texture);
+        }
+    }
+}
+
+
+
+// ============
+// === Line ===
+// ============
+
+/// A structure keeping line of glyphs with proper alignment.
+///
+/// Not all the glyphs in `glyphs` vector may be actually in use; this structure is meant to keep
+/// changing text, and for best performance it re-uses the created Glyphs (what means the specific
+/// buffer space). Therefore there is a cap for line length. See also `GlyphSystem::new_empty_line`.
+pub struct Line {
+    glyphs         : Vec<Glyph>,
+    baseline_start : Vector2<f32>,
+    base_color     : Vector4<f32>,
+    height         : f32,
+    font_id        : FontId,
+}
+
+impl Line {
+    /// Replace currently visible text.
+    ///
+    /// The replacing strings will reuse glyphs which increases performance of rendering text.
+    pub fn replace_text<Chars>(&mut self, chars:Chars, fonts:&mut FontRegistry)
+    where Chars : Iterator<Item=char> + Clone {
+        let font = fonts.get_render_info(self.font_id);
+        let pen  = PenIterator::new(self.baseline_start,self.height,chars.clone(),font);
+
+        for (glyph,(_,position)) in self.glyphs.iter_mut().zip(pen) {
+            glyph.set_position(Vector3::new(position.x,position.y,0.0));
+        }
+        for (glyph,ch) in self.glyphs.iter_mut().zip(chars) {
+            let font       = fonts.get_render_info(self.font_id);
+            let glyph_info = font.get_glyph_info(ch);
+            let size       = glyph_info.scale.scale(self.height);
+            let offset     = glyph_info.offset.scale(self.height);
+            // Pen position is on bottom-left corner of glyph, but sprites are positioned by its
+            // center.
+            let offset     = offset + size.scale(0.5);
```

Comment:
Could you please add option to sprite system which will change the alignment of sprites to a corner if needed instead? (it sohuld change their geometry material).